lib/styling-service.js

```diff
@@ -16,8 +16,7 @@ class StylingService {
     }
 
     static getTempFilename(book) {
-        const extname = path.extname(book.getCoverPath());
-        return `${book.getId()}${extname}`;
+        return `${book.getId()}.png`;
```

Comment:
If the image is a `.png` - would it also work to just rename it to `.png`?
This method previously would have worked with any filepaths and now it's going to make every path a `.png`. 
If there's a need to convert everything to a `.png` - maybe that could be an explicit step/option?

Reply:
Based on this:
> Cover images are included (must be in PNG format). I recommend 600x800 or a larger alternative of the same aspect ratio.

(https://github.com/kcartlidge/nodepub)


I would say that the final cover that gets embedded into the .epub *has to* be in PNG format. That's why I changed the output filename to explicitly say that it is `.png`. I cannot be in any format because nodepub embeds it as `cover.png` and specifies its content type to be `imge/png`.
See:
* https://github.com/kcartlidge/nodepub/blob/356b725c83cafe851550f5e300553f07b6b70124/index.js#L79 
* and https://github.com/kcartlidge/nodepub/blob/356b725c83cafe851550f5e300553f07b6b70124/constituents/structural.js#L60

It does not really matter what is the input format of the cover - it still can be `.jpg` as your case: `assets/cover.jpg` - what matters is the output format that jimp will use when writing the file - at it determines that based on the output filename extension.

Reply:
Makes sense to have the cover be `.png`. It also sounded like in your issue that `cover.jpg` is already a `.png`?

In that case shouldn't it also work to:

1. Rename `assets/cover.jpg` to `assets/cover.png`.
1. Update `lib/config.js` to have `Config.DEFAULT_COVER_PATH = `${Config.ASSETS_PATH}/cover.png`.

I would prefer changing the file we're passing in to have the correct file extension rather than assume all passed in files are going to have `.png`. I've tested that out locally and it seems to work?

Reply:
No no, let me start over :)
Here's how it works WITHOUT the fix:

1. `assets/cover.jpg` is *really* in JPEG format:
```
cover.jpg: JPEG image data, JFIF standard 1.01, aspect ratio, density 1x1, segment length 16, progressive, precision 8, 816x1056, frames 3
```
2. `jimp` loads the image from that file.
3. `getTempFilename()` generates a new (temporary) file name for the generated custom cover. It preserves the `.jpg` extension. Due to how `jimp` works, it *also* saves the custom cover in JPEG format - the extension determines what output format should be used.
4. `nodepub` receives the path to that temporary file, reads its **content** (image in JPEG format) and writes this inside `.epub`as `images/cover.png`. It assumes the image is in PNG format, as it does not support any other format.
5. Most e-Book readers don't mind that the inside the `.epub` it says the image is in PNG format, while it is really a JPEG - as long as it is in a supported image format.
6. PocketBook readers try to read `images/cover.png` as it really were a PNG. And fail miserably.

WITH the fix:

1. `assets/cover.jpg` can still be in JPEG format as it is now. Or any other format that `jimp` suppports. It can be in `.png` but it does not really matter.
2. `jimp` reads the image from that file.
3. `getTempFilename()` generates a new (temporary) file name for the generated custom cover. It **does not** preserve the input extension, but rather it enforces it to be `.png`
4. `nodepub` still assumes that the file is in PNG format, but now that assumption is correct.
5. PocketBook readers are happy :)

Sorry if I was not able to explain that clear enough right from the start.

Reply:
Thanks for the clear steps! So you're saying that jimp does the conversion from `.jpg` -> `.png` because the filename returned is `.png`? If the input image is `.jpg` and when it comes out it's `.png` -> sounds like something is doing the conversion?

I think I'm starting to understand better. 😅

Reply:
Yes. Precisely. JIMP does the conversion if we tell him that the output file is supposed to be .png. So that's what we need to do here: tell JIMP (by producing appropriate filename in getTempFilename()) that what we want for the final image is to have it in the PNG format.

Reply:
Got it! Thanks for the explanation. Makes a lot of sense now!